File: src/main/java/it/astromark/agenda/schoolclass/dto/TeachingTimeslotDetailedResponse.java
package it.astromark.agenda.schoolclass.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import it.astromark.classwork.dto.ClassActivityResponse;
import it.astromark.classwork.dto.HomeworkBriefResponse;
import it.astromark.user.commons.dto.SchoolUserResponse;

@Schema(description = "Response object containing detailed information about a teaching timeslot")
public record TeachingTimeslotDetailedResponse(
        Integer id,
        Integer hour,
        String subject,
        Boolean signed,
        SchoolUserResponse teacher,
        ClassActivityResponse activity,
        HomeworkBriefResponse homework
) {
}

File: src/main/typescript/src/entities/TeachingTimeslotDetailedResponse.ts
import {DateObject} from "react-multi-date-picker";

export type TeachingTimeslotDetailedResponse = {
    id: number | null;
    hour: number;
    subject: string;
    signed: boolean;
    teacher: SchoolUserResponse;
    activity: ClassActivityResponse | null;
    homework: HomeworkBriefResponse | null;
}

export type SchoolUserResponse = {
    id: string;
    name: string;
    surname: string;
}

export type ClassActivityResponse = {
    id: number;
    title: string;
    description: string;
}

export type HomeworkBriefResponse = {
    id: number;
    title: string;
    description: string;
    dueDate: DateObject;
    hasChat: boolean;
}